### ISSUE TITLE
error notifying systemd

### ISSUE THREAD

**Issue by dataO1** [hashicorp/vault#15532]
**Describe the bug**
The vault systemd service gets permission denied when trying to write to systemd-notify and fails to start. Here are the relevant logs:
```
...
vault[7909]: ==> Vault server started! Log data will stream in below:
vault[7909]: 2022-05-20T13:49:51.761+0200 [INFO]  proxy environment: http_proxy="" https_proxy="" no_proxy=""
vault[7909]: 2022-05-20T13:49:51.773+0200 [INFO]  core: Initializing versionTimestamps for core
vault[7909]: 2022-05-20T13:49:51.774+0200 [ERROR] error notifying systemd: error="write unixgram @->/run/systemd/notify: write: permission denied
```

**To Reproduce**
1. Update to 1.10.3
2. Start the vault systemd service

**Expected behavior**
The vault service starts.

**Environment:**
* Vault Server Version (retrieve with `vault status`): 1.10.3
* Vault CLI Version (retrieve with `vault version`): 1.10.3
* Server Operating System/Architecture: AlmaLinux/x86

Vault server configuration file(s):

```hcl

ui = true
api_addr = "https://xxx:8200"

storage "file" {
  path = "/opt/vault/data"
}

# HTTPS listener
listener "tcp" {
  address       = "xxx:8200"
  tls_cert_file = "/etc/vault.d/certs/xxx.cert"
  tls_key_file  = "/etc/vault.d/certs/xxx.key"
  tls_min_version = "tls13"
}

plugin_directory = "/etc/vault.d/plugins"
```

**Additional context**
For a temporary fix i removed `Type=notify` from the sytemd unit file. After that vault works as expected.

**Comment by cipherboy** [comment]
@dataO1 Out of curiosity, are you running [SELinux policy](https://github.com/hashicorp/vault-selinux-policies)? Note that Vault is operating fine (we log, not err out on failure to notify systemd), but our inability to notify systemd is causing systemd to consider the Vault service failed.

This makes us think perhaps this might be an issue in your environment. Perhaps the [Vault discussion forum](https://discuss.hashicorp.com/c/vault/30) is a better venue?

**Comment by dataO1** [comment]
Yes, we use the SElinux policies, also there are no SELinux policy violations, when checking `ausearch -c 'vault'`.
Before the update, vault was running as expected. When removing the `Type=notify` entry in the systemd unit vault works, which indicates that it's not an issue in the environment i think.

> Note that Vault is operating fine ([we log](https://github.com/hashicorp/vault/blob/08693a2950d20115bfdac35a922597c8f116facc/command/server.go#L1656-L1667), not err out on failure to notify systemd), but our inability to notify systemd is causing systemd to consider the Vault service failed.

I see, do you know of some way to tell systemd to keep the process alive, even if not being able to communicate to notify?

**Comment by maxb** [comment]
> When removing the `Type=notify` entry in the systemd unit vault works, which indicates that it's not an issue in the environment i think.

I don't agree with this deduction - what the permissions are on the systemd notify socket, is not something controlled by the Vault binary.

Please show your systemd unit file for Vault, which seems like the next logical thing to look at.



> I see, do you know of some way to tell systemd to keep the process alive, even if not being able to communicate to notify?

Only by turning notify support off entirely by removing `Type=notify`.

**Comment by dataO1** [comment]
Here's the unit file (it comes from the rpm repo, see [here](https://www.hashicorp.com/blog/announcing-the-hashicorp-linux-repository)):

```
[Unit]
Description="HashiCorp Vault - A tool for managing secrets"
Documentation=https://www.vaultproject.io/docs/
Requires=network-online.target
After=network-online.target
ConditionFileNotEmpty=/etc/vault.d/vault.hcl
StartLimitIntervalSec=60
StartLimitBurst=3

[Service]
EnvironmentFile=/etc/vault.d/vault.env
User=vault
Group=vault
ProtectSystem=full
ProtectHome=read-only
PrivateTmp=yes
PrivateDevices=yes
SecureBits=keep-caps
AmbientCapabilities=CAP_IPC_LOCK
CapabilityBoundingSet=CAP_SYSLOG CAP_IPC_LOCK
NoNewPrivileges=yes
ExecStart=/usr/bin/vault server -config=/etc/vault.d/vault.hcl
ExecReload=/bin/kill --signal HUP $MAINPID
KillMode=process
KillSignal=SIGINT
Restart=on-failure
RestartSec=5
TimeoutStopSec=30
LimitNOFILE=65536
LimitMEMLOCK=infinity

[Install]
WantedBy=multi-user.target
```

**Comment by aphorise** [comment]
hey @dataO1 - this seem to be something specific to your environment and I'm curious:
 - if you've pursued SELinux: `setools`, `settroubleshoot` & `/var/log/audit/audit.log` or if they show any `denied` or other possibly clues?
 - the last systemd unit file above that you've shared is what works for you?
 - How relevant is this issue for you still and what do you see as next?

**Comment by dataO1** [comment]
Hi,

 - no SELinux or permission errors which i could find.
 - yes, thats the one that worked (without the Type=notify)
 - Honestly not very relevant, i just found this and was wondering whats going on. I understand if this is not a priority to you.

**Comment by aphorise** [comment]
@dataO1 thanks for the confirmation - are you okay to proceed to close?

**Comment by ProbstenHias** [comment]
@dataO1 I am currently running in the same problem as you.
Would you mind telling me how you managed to fix this?
